show diffs for stash items

campersau opened the pull request:
reference #444

Clicking on the stash title will show the changed files.

![image](https://cloud.githubusercontent.com/assets/4009570/14227660/03a6b634-f903-11e5-861b-71adf35db5a3.png)

![image](https://cloud.githubusercontent.com/assets/4009570/14227647/afe5efc4-f902-11e5-8e05-3ff53222ca08.png)

campersau commented:
@codingtwinky wouldn't it be better to move the `Default` and `Side by Side` buttons and the `textDiffType` logic to the `commitdiff` component?

Because these things are now duplicated in the `stash` and `commit` component.

FredrikNoren commented:
Really nice! Only two comments from me are the usual suspects: add to CHANGELOG and, if we can, a basic click-test so it doesn't break in the future

codingtwinky commented:
@campersau I do like the thought and if we do that we can reduce duplicate logic.  we can also further remove same duplicate logic via recycling comittdiff component and use it in staging, right now there are many duplicate logics between diff for staging and diff for commits as code base is different. 

If you would like to do that with in this PR please let me know so I can pause #725 as i'd imagine there will be many conflicts.

campersau commented:
@codingtwinky okay lets do this after this PR and #725 are merged.

codingtwinky commented:
@campersau #725 is merged, have you had the chance to work on this yet?  Is there anything I can do to help?  I think this would be nice...

campersau commented:
@codingtwinky I will try to work on this this week.

campersau commented on components/staging/staging.html:
```diff
@@ -49,20 +49,20 @@
+            <button class="btn btn-default bootstrap-tooltip" data-ta-clickable="stage-wordwrap" data-bind="click: toggleWordWrap.bind($data, true), css: {active: wordWrap}" data-toggle="tooltip" data-placement="bottom" data-original-title="Wrap words per line" data-delay='{"show":"2000", "hide":"0"}'>
+              <span data-bind="text: 'Word Wrap'"></span>
+            </button>
           </div>
```
I just moved these things around to be consistent with the other buttons. (Show `Default` as first option.)

campersau commented:
I think this is good to go now:
- moved the `Default` and `Side by Side` buttons and the `textDiffType` logic to the `commitdiff` component
- added clicktests for the `stash` component. stash -> open diff -> pop stash

codingtwinky commented:
:+1:

Thanks for moving the toggle button logic to commit diff!
